Update octavia.rst (#1362)
stackhpc/stackhpc-kayobe-config: doc/source/operations/octavia.rst

Octavia
=======

.. Introduction to Octavia:

Introduction to Octavia
=======================

Octavia is known as an *operator-scaled* load balancing service, whose primary goal is to
distribute incoming network traffic, destined for a single VM, across multiple VMs instead;
thus preventing overloading the VM and improving the overall performance of the VM's outbound
service.

For more information on Octavia, please refer to the `official documentation <https://docs.openstack.org/octavia/latest/reference/introduction.html>`_.

.. _Deploying Octavia:

> Deploying Octavia

Reviewer: Could you restructure this into a sequence of individual steps? See for example the [Multinode page](https://stackhpc-kayobe-config.readthedocs.io/en/stackhpc-2023.1/contributor/environments/ci-multinode.html#multinode-test-environment) which goes through a few things step-by-step

Author: This seems a little unnecessary for this because it would end up being 2 steps; this is more to make people aware of the various configs which could snag the deployment.

Reviewer: I think at the moment it misses a few important steps, in particular "It of course goes without saying that the network should exist in networks.yml" is a pretty big thing to gloss over.

So for example you could have:
1. Where to enable the service in kolla.yml
2. How to define the octavia network in networks.yml
3. How to define network interfaces for controllers
4. What to check for neutron provider networks
5. How to select the right driver(s), and the differences between Amphora vs OVN
6. How to properly configure the Amphora driver

Reviewer: Please also keep in mind that simple, clear instructions make for the best docs. try to stay away from _wordyness_.

For example:

> Much like any other Kolla managed service, the method of deploying Octavia is as simple
as ...

could be written as:

> To enable Octavia, set ...

Author: I have an idea on how to make a nice comprehensive list. Regarding the *wordyness*, in some cases it is intentional, in the example provided mentioning its similarity to other Kolla managed services will give users a lot of context as to what to expect, as long as they have deployed at least one service before; in addition to reinforcing the steps for deploying services. In other cases, I agree and some sentences are wordy without adding anything.

=================

Much like any other Kolla managed service, the method of deploying Octavia is as simple
as enabling ``kolla_enable_octavia: true`` within the chosen environment's ``kolla.yml``.
However, before running a ``kayobe overcloud service configure``, check
``octavia_net_interface`` (often found within the ``controllers`` directory within
``${KAYOBE_CONFIG_PATH}/inventory/group_vars``) to see the network which Octavia management
network will be using. If it is configured to use a VLAN then ``kolla_enable_neutron_provider_networks: true``
should also be set in ``kolla.yml``.

.. note::

      It of course goes without saying that the network configured for ``octavia_net_interface``
      should also exist in ``networks.yml``.

> By default Octavia will deploy an Amphorae (a single Amphora VM) per load balancing service, this should be

Reviewer 2: Amphorae is the plural of Amphora.
```suggestion
By default Octavia will deploy an Amphora (a single Ubuntu VM running HAProxy) per load balancing service. This should be
```

Author: It is, indeed, don't know why I changed it originally.

changed to make them highly available and have two Amphora VM per service. Done by setting
``octavia_loadbalancer_topology: "ACTIVE_STANDBY"`` in ``${KAYOBE_CONFIG_PATH}/kolla/globals.yml``,
this will ensure that if the master Amphora VM were to go down, the other would be able to

> take over the load balancing functions.

Reviewer: Only applies when using the Amphora driver

Author: Made this clearer.

Further configuration options and details on installation can be found in the

> `Octavia documentation <https://docs.openstack.org/octavia/latest/install/index.html>`_.

Reviewer: The kolla-ansible documentation will be more relevant here than the raw Octavia documentation
```suggestion
`Octavia documentation <https://docs.openstack.org/kolla-ansible/latest/reference/networking/octavia.html>`_.
```

Author: Agreed

   - IF NOT CONFIGURED  

     2.1. Check if a ``bond_interface`` has been configured, still within ``network-interfaces.yml``.  

     2.2. Check whether other network interfaces, such as ``internal_interface``, are configured to use ``{{ bond_interface }}`` and/or ``{{ .._vlan }}``.  
     
     2.3. If they are, then ``octavia_net_interface: "{{ brbond0_interface }}.{{ octavia_net`` ± ``_vlan }}"``.  

   - IF CONFIGURED   

     2.1. Continue to step 3.

#. Check that the ``{{ .._net_.. }}`` network configured for ``octavia_net_interface`` exists in ``networks.yml``.  

   - IF NOT CONFIGURED  

     3.1. Set ``octavia_net_name: octavia_net``.  
     
     3.2. Configure the Octavia network IP information, making sure to set ``octavia_net_vlan`` if using a VLAN.  
   
   - IF CONFIGURED  
     
     3.1. Continue to step 4.

#. Dependencies if:  

   - USING VLAN  

     4.1. Set ``kolla_enable_neutron_provider_networks: true`` in ``kolla.yml``.  
   
   - USING AMPHORA  
  
     4.1. Set ``octavia_loadbalancer_topology: "ACTIVE_STANDBY"`` in ``${KAYOBE_CONFIG_PATH}/kolla/globals.yml``.

> #. Run ``kayobe overcloud service configure``.

Reviewer 2: ```suggestion
#. Run ``kayobe overcloud service reconfigure``.
```
